packages/models-library/src/models_library/api_schemas_storage.py

```diff
@@ -86,107 +91,114 @@ class Config:
-
-
 # /locations/{location_id}/files/metadata:
 # /locations/{location_id}/files/{file_id}/metadata:
-class FileMetaData(BaseModel):
-    file_uuid: Optional[str] = Field(
-        description="Unique identifier for a file, like bucket_name/project_id/node_id/file_name = /bucket_name/object_name",
+class FileMetaDataGet(BaseModel):
+    # Used by frontend
+    file_uuid: str = Field(
+        description="NOT a unique ID, like (api|uuid)/uuid/file_name or DATCORE folder structure",
     )
-
-    user_id: Optional[int]
-    user_name: Optional[str]
-
-    location_id: Optional[int] = Field(description="Storage location")
-    location: Optional[str] = Field(description="Storage location display name")
-
-    bucket_name: Optional[str] = Field(description="Name of the s3 bucket")
-    object_name: Optional[str] = Field(
-        description="Name of the s3 object within the bucket"
+    location_id: LocationID = Field(..., description="Storage location")
+    project_name: Optional[str] = Field(
+        default=None,
+        description="optional project name, used by frontend to display path",
     )
-
-    project_id: Optional[UUID]
-    project_name: Optional[str]
-    node_id: Optional[UUID]
-    node_name: Optional[str]
-    file_name: Optional[str] = Field(description="Display name for a file")
-
-    file_id: Optional[str] = Field(
-        description="Unique uuid for the file. For simcore.s3: uuid created upon insertion and datcore: datcore uuid",
+    node_name: Optional[str] = Field(
+        default=None,
+        description="optional node name, used by frontend to display path",
     )
-    raw_file_path: Optional[str] = Field(description="Raw path to file")
-    display_file_path: Optional[str] = Field(
-        description="Human readlable  path to file"
+    file_name: str = Field(..., description="Display name for a file")
+    file_id: StorageFileID = Field(
+        ...,
+        description="THIS IS the unique ID for the file. either (api|project_id)/node_id/file_name.ext for S3 and N:package:UUID for datcore",
     )
-
-    created_at: Optional[datetime]
-    last_modified: Optional[datetime]
-    file_size: Optional[int] = Field(-1, description="File size in bytes")
-    entity_tag: Optional[str] = Field(
-        description="Entity tag (or ETag), represents a specific version of the file",
+    created_at: datetime
+    last_modified: datetime
+    file_size: ByteSize = Field(-1, description="File size in bytes (-1 means invalid)")
+    entity_tag: Optional[ETag] = Field(
+        default=None,
+        description="Entity tag (or ETag), represents a specific version of the file, None if invalid upload or datcore",
     )
     is_soft_link: bool = Field(
         False,
         description="If true, this file is a soft link."
         "i.e. is another entry with the same object_name",
     )
 
-    parent_id: Optional[str]
+    @validator("location_id", pre=True)
+    @classmethod
+    def convert_from_str(cls, v):
```

Comment:
MINOR: more like enforce_int. 
Q: why not making ``LocationID = int``?

Reply:
yes I tried being smart by setting it as literal. in the next PR I will instead do a Enum (which makes sense right?) and the conversion will be a given.